dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -1934,14 +1934,22 @@ bool
 DomainParticipantImpl::is_clean() const
 {
   bool sub_is_clean = subscribers_.empty();
-  bool topics_is_clean = topics_.size() == 0;
+  bool topics_is_clean = true;
+  int non_bit_topic_count = 0;
+
+  // check that the only remaining topics are built-in topics
+  for (TopicMap::const_iterator it = topics_.begin(); it != topics_.end(); ++it) {
+    if (!topicIsBIT(it->second.pair_.svt_->get_name(),it->second.pair_.svt_->get_type_name())) {
```

Comment:
this leaks memory from the strings

Reply:
The change you made works.  The other way to deal with it is `TopicImpl::type_name` instead of `get_type_name`, then the similar method would have to be added for `name`, I don't think it's there yet.

Reply:
I've added a TopicImpl::topic_name() which avoids extra string copies in is_clean().